### PR TITLE
feat(pie-button): DSW-1027 add destructive and destructive-ghost button variants

### PR DESCRIPTION
**Pull request by xander-marjoram**
## Describe your changes (can list changeset entries if preferable)
- [Added] - destructive and destructive-ghost button variants
- [Fixed] - hover and active colours for ghost-inverse variant
- [Added] - mixin for reassigning button spinner HSL custom properties

## Author Checklist (complete before requesting a review)
- [x] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests
- [ ] If it is a `PIE Docs` change, I have reviewed the Docs site preview
- [x] If it is a component change, I have reviewed the Storybook preview
- [ ] If there are visual test updates, I have reviewed them properly before approving

### Reviewer 1
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### Reviewer 2
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### CODE REVIEW

File: packages/components/pie-button/src/button.scss [hunk 1]
```diff
@@ -6,18 +6,44 @@
...
+
+        --btn-bg-color: hsl(var(#{$bg-color}-h), var(#{$bg-color}-s), calc(var(#{$bg-color}-l) + var(--active-modifier)));
     }
 }
 
+/**
+* Shorthand mixin for updating the HSL custom properties for the loading spinner.
+* Takes in the name of the color to be split into its HSL components.
+*/
+@mixin spinner-base-colors($color) {
```
Comment (jamieomaguire) [inline packages/components/pie-button/src/button.scss:41]: 🔥

File: packages/components/pie-button/src/defs.ts [hunk 2]
```diff
@@ -1,6 +1,8 @@
 export const sizes = ['xsmall', 'small-productive', 'small-expressive', 'medium', 'large'] as const;
 export const types = ['submit', 'button', 'reset', 'menu'] as const;
-export const variants = ['primary', 'secondary', 'outline', 'ghost', 'inverse', 'ghost-inverse'] as const;
+export const variants = [
```
Comment (jamieomaguire) [inline packages/components/pie-button/src/defs.ts:3]: you just need to update the readme with the new prop values and this looks good to go!

File: apps/pie-storybook/stories/pie-button.stories.ts [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import { html, TemplateResult } from 'lit';
-import type { StoryObj as Story } from '@storybook/web-components';
+import { type StoryObj as Story } from '@storybook/web-components';
```
Comment (raoufswe) [inline apps/pie-storybook/stories/pie-button.stories.ts:2]: just curious, what's the difference between both? does it make sense to update the generator to use this intead?

File: packages/components/pie-button/src/button.scss [hunk 4]
```diff
@@ -6,18 +6,44 @@
...
-@mixin button-interactive-states($bg-color) {
+/**
+* Defines button interactive states for hover, active and loading.
+* Takes the interactive color name as an argument, and whether the color
+* should be lightened or darkened.
+*/
+@mixin button-interactive-states($bg-color, $lighten: false) {
     &:hover:not(:disabled) {
-        background-color: hsl(var(#{$bg-color}-h), var(#{$bg-color}-s), calc(var(#{$bg-color}-l) - var(--dt-color-hover-01)));
+        --hover-modifier: calc(-1 * var(--dt-color-hover-01));
```
Comment (fernandofranca) [inline packages/components/pie-button/src/button.scss:16]: For curiosity (and learning) sake, why to multiply the color by -1?

**Review by fernandofranca — DISMISSED** [review]

File: apps/pie-storybook/stories/pie-button.stories.ts (hunk 3 above)
Comment (xander-marjoram) [reply apps/pie-storybook/stories/pie-button.stories.ts:2]: I don't actually remember making this change, it was either an automatic change or from a merge update. I _think_ they're both equivalent.

**Review by kevinrodrigues — APPROVED** [review]

File: packages/components/pie-button/src/button.scss (hunk 4 above)
Comment (fernandofranca) [reply packages/components/pie-button/src/button.scss:16]: Awesome! Thanks for the explanation!

**Review by fernandofranca — APPROVED** [review]

File: apps/pie-storybook/stories/pie-button.stories.ts (hunk 3 above)
Comment (raoufswe) [reply apps/pie-storybook/stories/pie-button.stories.ts:2]: cool i will test it and see if it's needed.

**Review by raoufswe — APPROVED** [review]

**Pull request merged**